clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -1232,6 +1230,12 @@ RValue CIRGenFunction::buildCall(clang::QualType CalleeType,
       Addr = Fn->getResult(0);
     }
 
+    // Use the same addr space as Addr
+    auto CalleePtrAS =
+        Addr.getType().cast<mlir::cir::PointerType>().getAddrSpace();
+    auto CalleePtrTy = mlir::cir::PointerType::get(builder.getContext(),
+                                                   CalleeTy, CalleePtrAS);
```

Comment:
~Do you think you can hide the `mlir::cir::PointerType::get(builder.getContext(),` part in a `CIRGenBuilder` (or `CIRBaseBuilder`) method with a name along `getPointerTy`?~

Can you use `getPointerTo` (from `CIRBaseBuilder`) instead? Hides some of the horrors ;)